assessmentModel/src/androidLibMain/kotlin/android.kt

```diff
@@ -28,8 +30,19 @@ actual object UUIDGenerator {
     actual fun uuidString() : String = UUID.randomUUID().toString()
 }
 
-actual object DateGenerator {
-    actual fun nowString(): String = ZonedDateTime.now().format(DateTimeFormatter.ISO_DATE_TIME)
+actual object DateUtils {
+    actual fun nowString(): String = ZonedDateTime.now().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
     
     actual fun currentYear(): Int = ZonedDateTime.now().year
+
+    actual fun bridgeIsoDateTimeString(instant: Instant): String {
+        val jtInstant = java.time.Instant.ofEpochMilli(instant.toEpochMilliseconds())
+        return ZonedDateTime.ofInstant(jtInstant, ZoneId.systemDefault()).format(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
```

Comment:
## Problem 1

According to the Java docs, this does *not* include fractional seconds. Are they wrong? Does getting the instant in milliseconds automagically add that?

I think milliseconds were included so that the format would be consistent across all use-cases including for recorders that sample more than once per second.

From the Java docs:
ISO_OFFSET_DATE_TIME | Date Time with Offset | '2011-12-03T10:15:30+01:00'
-- | -- | --

## Problem 2
`ZoneId.systemDefault()` doesn't allow for unit testing b/c the tests would only work in the "America/Los_Angeles" timezone.

I would recommend allowing the timezone to be optionally passed. For example:
```
actual fun bridgeIsoDateTimeString(instant: Instant, zoneId: String = ZoneId.systemDefault() ): String
```

Or allow tests to set the "actual" timezone ID.

Reply:
**Problem 1**
The resulting string has fractional seconds.  Here is an example: 2021-03-25T10:53:59.471-07:00

**Problem 2**
Just pushed an update which takes an optional TimeZone parameter (kotlinx.datetime version of ZoneId).